client/app/app.routes.js

```diff
@@ -6,7 +8,8 @@ angular
       controller: 'AppCtrl',
       controllerAs: 'AppCtrl',
       resolve: {
-        currentUser: /* @ngInject */ User => User.getUser(),
+        currentUser: /* @ngInject */ User => User.getUser()
+          .then(user => new UserModel(user)),
```

Comment:
Nice :)

Reply:
Would have named it `User` though, it's an user after all

Reply:
@FredericEspiau `User` is the name of a `service` so to avoid confusion